homeassistant/components/roon/media_player.py

```diff
@@ -288,6 +301,16 @@ def update_state(self):
         self._attr_is_volume_muted = volume["muted"]
         self._attr_volume_step = volume["step"]
         self._attr_volume_level = volume["level"]
+        self.volume_fixed = volume["fixed"]
+        self.volume_incremental = volume["incremental"]
+        if not self.volume_fixed:
+            self._attr_supported_features = (
+                self._attr_supported_features | MediaPlayerEntityFeature.VOLUME_STEP
+            )
+            if not self.volume_incremental:
+                self._attr_supported_features = (
+                    self._attr_supported_features | MediaPlayerEntityFeature.VOLUME_SET
+                )
```

Comment:
Does this need to be done every time the state is updated, or can we do it once, preferably in `__init__`?

Reply:
The media_player number and state are updated in an async way by roon, so this can change as endpoints are enabled and disabled (this can be as simple as powering on a USB dac - which will enable the connected endpoint). 

The volume type can also be changed by the user in roon - so I think it is safest to leave this as updating on every update, since that way it will never be wrong!